cardano-api/src/Cardano/Api/MetaData.hs

```diff
@@ -22,13 +25,101 @@ import qualified Data.Map.Strict as Map
 import qualified Data.Scientific as Scientific
 import qualified Data.Text as Text
 import qualified Data.Text.Encoding as Text
-
 import qualified Data.Vector as Vector
 
 import           Shelley.Spec.Ledger.MetaData (MetaData (..), MetaDatum (..))
 
 
 
+-- | A transaction metadata validation error.
+data TxMetadataValidationError
+  = TxMetadataTextStringInvalidLengthError
+    -- ^ The length of a text string metadatum value exceeds the maximum.
+      !Int
+      -- ^ Maximum byte length.
+      !Int
+      -- ^ Actual byte length.
+  | TxMetadataByteStringInvalidLengthError
+    -- ^ The length of a byte string metadatum value exceeds the maximum.
+      !Int
+      -- ^ Maximum byte length.
+      !Int
+      -- ^ Actual byte length.
+  deriving (Eq, Show)
+
+-- | The maximum byte length of a transaction metadata text string value.
+txMetadataTextStringMaxLength :: Int
+txMetadataTextStringMaxLength = 64
+
+-- | The maximum length of a transaction metadata byte string value.
+txMetadataByteStringMaxLength :: Int
+txMetadataByteStringMaxLength = 64
+
+renderTxMetadataValidationError :: TxMetadataValidationError -> Text
+renderTxMetadataValidationError err =
+  case err of
+    TxMetadataTextStringInvalidLengthError maxLen actualLen ->
+      "Text string metadatum value must consist of at most "
+        <> show maxLen
+        <> " bytes, but it consists of "
+        <> show actualLen
+        <> " bytes."
+    TxMetadataByteStringInvalidLengthError maxLen actualLen ->
+      "Byte string metadatum value must consist of at most "
+        <> show maxLen
+        <> " bytes, but it consists of "
+        <> show actualLen
+        <> " bytes."
+
+-- | Validate the provided transaction metadata.
+validateTxMetadata :: TxMetadata
+                   -> Either TxMetadataValidationError TxMetadata
+validateTxMetadata txMd@(TxMetadata (MetaData mdMap)) =
+    Map.foldl' foldFn (Right txMd) mdMap
+  where
+    foldFn
+      :: Either TxMetadataValidationError TxMetadata
+      -> MetaDatum
+      -> Either TxMetadataValidationError TxMetadata
+    foldFn lastResult metaDatum =
+      case lastResult of
+        Left err -> Left err
+        Right _ -> validate metaDatum
```

Comment:
I think this is essentially the same as `second (const . validate metaDatum) lastResult` or `either Left (const . validate metaDatum) lastResult`.